pandas/tests/series/test_duplicates.py

```diff
@@ -0,0 +1,142 @@
+# coding=utf-8
+# pylint: disable-msg=E1101,W0612
+
+import pytest
+
+import numpy as np
+import pandas as pd
+
+from pandas import Series
+
+from pandas.util.testing import assert_series_equal
+import pandas.util.testing as tm
+from .common import TestData
+
+
+class TestSeriesDuplicates(TestData):
+
+    def test_value_counts_nunique(self):
+
+        # basics.rst doc example
+        series = Series(np.random.randn(500))
+        series[20:500] = np.nan
+        series[10:20] = 5000
+        result = series.nunique()
+        assert result == 11
+
+        # GH 18051
+        s = pd.Series(pd.Categorical([]))
+        assert s.nunique() == 0
+        s = pd.Series(pd.Categorical([np.nan]))
+        assert s.nunique() == 0
+
+    def test_unique(self):
+
+        # 714 also, dtype=float
+        s = Series([1.2345] * 100)
+        s[::2] = np.nan
+        result = s.unique()
+        assert len(result) == 2
+
+        s = Series([1.2345] * 100, dtype='f4')
+        s[::2] = np.nan
+        result = s.unique()
+        assert len(result) == 2
+
+        # NAs in object arrays #714
+        s = Series(['foo'] * 100, dtype='O')
+        s[::2] = np.nan
+        result = s.unique()
+        assert len(result) == 2
+
+        # decision about None
+        s = Series([1, 2, 3, None, None, None], dtype=object)
+        result = s.unique()
+        expected = np.array([1, 2, 3, None], dtype=object)
+        tm.assert_numpy_array_equal(result, expected)
+
+        # GH 18051
+        s = pd.Series(pd.Categorical([]))
+        tm.assert_categorical_equal(s.unique(), pd.Categorical([]),
+                                    check_dtype=False)
+        s = pd.Series(pd.Categorical([np.nan]))
+        tm.assert_categorical_equal(s.unique(), pd.Categorical([np.nan]),
+                                    check_dtype=False)
+
+    def test_unique_data_ownership(self):
+        # it works! #1807
+        Series(Series(["a", "c", "b"]).unique()).sort_values()
+
+    def test_is_unique(self):
+        # GH11946
+        s = Series(np.random.randint(0, 10, size=1000))
+        assert not s.is_unique
+        s = Series(np.arange(1000))
+        assert s.is_unique
+
+    def test_is_unique_class_ne(self, capsys):
+        # GH 20661
+        class Foo(object):
+            def __init__(self, val):
+                self._value = val
+
+            def __ne__(self, other):
+                raise Exception("NEQ not supported")
+
+        li = [Foo(i) for i in range(5)]
+        s = pd.Series(li, index=[i for i in range(5)])
+        _, err = capsys.readouterr()
+        s.is_unique
+        _, err = capsys.readouterr()
+        assert len(err) == 0
+
+    @pytest.mark.parametrize(
+        'keep, expected',
+        [
+            ('first', Series([False, False, False, False, True, True, False])),
+            ('last', Series([False, True, True, False, False, False, False])),
+            (False, Series([False, True, True, False, True, True, False]))
+        ])
```

Comment:
either use 2 classes for test definitions or don’t use them (preferred)

Reply:
same for frame

Reply:
Classes of tests? I just wrote them in the same manner as in `test_analytics` and then transferred them here. You want me to get rid of `TestSeriesDuplicates` and do class-less tests - do I get that correctly?

Reply:
yes that is the old format is much less readable

we only use classes if they r logical separations